common/decklist.cpp

```diff
@@ -726,12 +726,29 @@ bool DeckList::deleteNode(AbstractDecklistNode *node, InnerDecklistNode *rootNod
 void DeckList::updateDeckHash()
 {
     QStringList cardList;
-    for (int i = 0; i < root->size(); i++) {
+    bool isValidDeckList = true;
+    QSet<QString> hashZones, optionalZones;
+
+    hashZones << "main" << "side"; // Zones in deck to be included in hashing process
+    optionalZones << "tokens"; // Optional zones in deck not included in hashing process
+
+    for (int i = 0; i < root->size(); i++)
+    {
         InnerDecklistNode *node = dynamic_cast<InnerDecklistNode *>(root->at(i));
-        for (int j = 0; j < node->size(); j++) {
+        for (int j = 0; j < node->size(); j++)
+        {
             DecklistCardNode *card = dynamic_cast<DecklistCardNode *>(node->at(j));
             for (int k = 0; k < card->getNumber(); ++k)
-                cardList.append((node->getName() == "side" ? "SB:" : "") + card->getName().toLower());
+            {
+                if (hashZones.contains(node->getName())) // Mainboard or Sideboard
```

Comment:
Shouldn't this block be moved up two loops? `node` is declared in the `i` loop but you're going to call it for every `j` and `k` in `i`